core/src/main/java/org/apache/iceberg/ManifestFiles.java

```diff
@@ -80,9 +98,26 @@ private ManifestFiles() {
    * @return a {@link ManifestReader}
    */
   public static ManifestReader<DataFile> read(ManifestFile manifest, FileIO io, Map<Integer, PartitionSpec> specsById) {
+    return read(manifest, io, EncryptionManagers.defaultManager(), specsById);
```

Comment:
I'd like to have compatibility with the bulk-decryption and bulk-encryption methods in the encryption manager. For bulk decryption, the `decrypt` method is passed `Iterable<EncryptedInputFile>` and returns `Iterable<InputFile>`. Then those input files need to be used.

I think that means we should refactor this a bit differently and create a `read(ManifestFile, InputFile, Map)` method that both this and the version with `EncryptionManager` call. Then this one could simply call `io.newInputFile(manifest.path())` and hand off to the `InputFile` reader. That avoids using the default manager where it isn't needed and allows us to use the bulk methods.